httpclient-core/src/main/java/esa/httpclient/core/AttributesMap.java

```diff
@@ -16,44 +16,77 @@
 package esa.httpclient.core;
 
 import esa.commons.Checks;
+import esa.commons.annotation.Internal;
 
 import java.util.Map;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 
-public class ContextImpl implements Context {
+@Internal
+public class AttributesMap {
 
-    private final Map<String, Object> attributes = new ConcurrentHashMap<>(0);
+    protected final Map<String, Object> attributes = new ConcurrentHashMap<>(4);
```

Comment:
init cap of 4 is a little small which would cause a frequent expansion, maybe we can use an init cap of 8.

Reply:
done